mlir/test/Target/LLVMIR/openmp-parallel-reduction-multiblock.mlir

```diff
@@ -0,0 +1,342 @@
+// RUN: mlir-translate -mlir-to-llvmir %s | FileCheck %s
+
+// generated by flang-new:
+// subroutine missordered_blocks(x,y)
+//   integer, allocatable :: x, y
+//   !$omp parallel reduction(+:x,y)
+//   x = 42
+//   y = 24
+//   !$omp end parallel
+// end subroutine
+
+// This is basically a test that we don't crash while translating this IR
+
+omp.declare_reduction @add_reduction_byref_box_heap_i32 : !llvm.ptr init {
```

Comment:
Please minimize this test as much as possible. Trivially running cse, canonicalize etc also possibly manually editing.

Reply:
This is an undesired property, as this is a massive maintenance burden for all changes to the LLVM IR export. Furthermore, this will run out of sync quickly, given that flang is modified, which is not great.

Tests for bugs should be as small as possible, while still covering the case the caused the crash. Also, reducing the number of checks is very much desired, especially when they cover something that is not relevant for the actual test.